packages/@aws-cdk/aws-certificatemanager/lib/dns_validated_certificate_handler/index.js

```diff
@@ -0,0 +1,196 @@
+ * The hosted zone ID must refer to a **public** Route53-managed DNS zone that is authoritative
+ * for the suffix of the certificate's Common Name (CN).  For example, if the CN is
+ * `*.example.com`, the hosted zone ID must point to a Route 53 zone authoritative
+ * for `example.com`.
+ *
+ * @param {string} requestId the CloudFormation request ID
+ * @param {string} domainName the Common Name (CN) field for the requested certificate
+ * @param {string} hostedZoneId the Route53 Hosted Zone ID
+ * @returns {string} Validated certificate ARN
+ */
+const requestCertificate = async function (requestId, domainName, hostedZoneId) {
+  const acm = new aws.ACM()
+  const route53 = new aws.Route53()
+  const encoder = new UuidEncoder('base36')
+
+  console.log(`Requesting certificate for ${domainName}`)
+
+  const reqCertResponse = await acm.requestCertificate({
+    DomainName: domainName,
+    IdempotencyToken: encoder.encode(requestId),
+    ValidationMethod: 'DNS'
+  }).promise()
+
+  console.log(`Certificate ARN: ${reqCertResponse.CertificateArn}`)
+
+  console.log('Waiting for ACM to provide DNS records for validation...')
+
+  var describeCertResponse
+  let attempt = 0
+  do {
+    // Exponential backoff based on 100ms base
+    await sleep(Math.pow(attempt, 2) * 100)
+    describeCertResponse = await acm.describeCertificate({
+      CertificateArn: reqCertResponse.CertificateArn
+    }).promise()
+  } while (describeCertResponse.Certificate.DomainValidationOptions[0].ValidationStatus !== 'PENDING_VALIDATION')
+
+  const record = describeCertResponse.Certificate.DomainValidationOptions[0].ResourceRecord
+
+  console.log(`Upserting DNS record into zone ${hostedZoneId}: ${record.Name} ${record.Type} ${record.Value}`)
+
+  const changeBatch = await route53.changeResourceRecordSets({
+    ChangeBatch: {
+      Changes: [{
+        Action: 'UPSERT',
+        ResourceRecordSet: {
+          Name: record.Name,
+          Type: record.Type,
+          TTL: 60,
+          ResourceRecords: [{
+            Value: record.Value
+          }]
+        }
+      }]
+    },
+    HostedZoneId: hostedZoneId
+  }).promise()
+
+  console.log('Waiting for DNS records to commit...')
+  await route53.waitFor('resourceRecordSetsChanged', {
+    // Wait up to 5 minutes
+    $waiter: {
+      delay: 30,
+      maxAttempts: 10
+    },
+    Id: changeBatch.ChangeInfo.Id
+  }).promise()
+
+  console.log('Waiting for validation...')
+  await acm.waitFor('certificateValidated', {
+    // Wait up to 5 minutes
+    $waiter: {
+      delay: 30,
+      maxAttempts: 10
+    },
+    CertificateArn: reqCertResponse.CertificateArn
+  }).promise()
+
+  return reqCertResponse.CertificateArn
+}
+
+/**
+ * Deletes a certificate from AWS Certificate Manager (ACM).
+ *
+ * @param {string} arn The certificate ARN
+ */
+const deleteCertificate = async function (arn) {
+  const acm = new aws.ACM()
+
+  console.log(`Deleting certificate ${arn}`)
+
+  await acm.deleteCertificate({
```

Comment:
Please handle the case if the certificate did not exist to start with.

Reply:
Will fix